### PR TITLE
Fix xml error in core/core.py

### PR DESCRIPTION
**Pull request by bushra2001**
**Issue:**

    • do_report function was passing list as an argument (json_obj) to exporter function instead of dictionary in 
      maryam/core/core.py. So due to absence of key value, it is unable to create xml tags. That is why it is displaying those 
      errors.
    • Character & is conflicting in xml file.

**Solution:**
    • I created an empty dictionary, and save key, value pair of required argument. In this way, this issue is resolved.
    • Replace & with &amp in json2xml function  in maryam/core/core.py 

I have tested these changes with following commands:

report xml dns_search osint/dns_search example.com
report xml docs osint/email_search microsoft.com
report xml wapps footprint/wapps owasp.org
report xml entry_points footprint/entry_points example.com
report xml social_nets osint/social_nets @owasp
report xml suggestion osint/suggest amazon
report xml crawl_pages footprint/crawl_pages owasp.org
report xml onion_search osint/onion_search amazon

report xml dns_search osint/dns_search 
report xml docs osint/email_search 
report xml wapps footprint/wapps 
report xml entry_points footprint/entry_points
report xml social_nets osint/social_nets
report xml suggestion osint/suggest 
report xml crawl_pages footprint/crawl_pages 
report xml onion_search osint/onion_search

Any suggestions will be highly appreciated

### CODE REVIEW

**Comment by z3y50n** [comment]
It doesn't work if I don't specify the query at the end of the command.
For example
`report xml doc osint/dns_search` 
doesn't work, but
`report xml doc osint/dns_search example.com`
works.

In the usage it clearly states that the `query` parameter is optional so maybe you can try and make it work even without it

File: core/core.py [hunk 1]
```diff
@@ -300,17 +299,17 @@ def save_gather(self, value, module, target, method=None, output=True):
...
 			return os.linesep.join(result_list)
 
-		if isinstance(json_obj_type, dict):
+		if isinstance(json_obj, dict):
 			for tag_name in json_obj:
 				sub_obj = json_obj[tag_name]
 				tag_name = re.sub(r"[\W]+", '_', tag_name)
 				result_list.append(f"{line_padding}\t<{tag_name}>")
-				result_list.append(self.json2xml(sub_obj, '\t' + line_padding))
+				result_list.append((self.json2xml(sub_obj, '\t' + line_padding).replace('&','&amp;')))
```
Comment (z3y50n) [inline core/core.py]: I think one set of parentheses is redundant

**Comment by bushra2001** [comment]
> It doesn't work if I don't specify the query at the end of the command.
> For example
> `report xml doc osint/dns_search`
> doesn't work, but
> `report xml doc osint/dns_search example.com`
> works.
> 
> In the usage it clearly states that the `query` parameter is optional so maybe you can try and make it work even without it

Thank you for pointing out the errors, I have fixed them. Kindly review it.

**Comment by z3y50n** [comment]
Everything looks good to me.
@saeeddhqan Shall I merge and close issue #90?

**Comment by saeeddhqan** [comment]
yes

**Comment by bushra2001** [comment]
It is been great working with you, thank you for your patience, and time. I want to take this project for GSoC. Looking forward to make many more contributions. :)

**Pull request merged**